nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-properties-loader/src/main/java/org/apache/nifi/properties/NiFiPropertiesLoader.java

```diff
@@ -289,4 +310,30 @@ private void setBootstrapKey(final SensitivePropertyProviderFactory providerFact
             }
         }
     }
+
+    private static class Props extends Properties {
+        // Only need to retain Properties key. This will help prevent possible inadvertent exposure of sensitive Properties value
+        private final Set<String> duplicateKeys = new HashSet<>();
+        public Props() {
+            super();
+        }
+
+        public Set<String> duplicateKeySet() {
+            return duplicateKeys;
+        }
+
+        @Override
+        public Object put(Object key, Object value) {
+            Object existingValue = super.put(key, value);
+            if (existingValue != null) {
+                if (existingValue.toString().equals(value.toString())) {
+                    logger.warn("Duplicate keys found for key '{}', but values are the same.", key);
```

Comment:
Recommend adjusting the message to remove the punctuation for easier reading:
```suggestion
                    logger.warn("Duplicate property key found [{}]", key);
```

Reply:
Wording updated